universal-application-tool-0.0.1/app/repository/DbRepository.java

```diff
@@ -0,0 +1,107 @@
+package repository;
+
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+
+import io.ebean.Ebean;
+import io.ebean.EbeanServer;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.Applicant;
+import models.Program;
+import models.Question;
+import play.db.ebean.EbeanConfig;
+
+public class DbRepository {
```

Comment:
question: should we add unit tests for this class? My interpretation of our testing strategy is that every Java class has unit tests, and functional tests are integration tests for multiple parts of the system (i.e. they actually inject multiple parts of the Play application) - please correct me if I'm on the wrong track

Reply:
Yeah, for all these existing methods we'd have to implement a mock database - which seems more trouble than it is worth, so I think we should hold off for now, and follow bion's suggestion if we ever have methods that do not involve db interop.